packages/experimental/Icon/src/Icon.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import {IconProps, SvgIconProps, FontIconProps} from './Icon.types';
+import * as ReactNative from 'react-native';
+import { Text } from '@fluentui-react-native/text';
+import {SvgUri} from 'react-native-svg'
+import * as assetRegistry from 'react-native/Libraries/Image/AssetRegistry';
+import { stagedComponent, useTheme, mergeProps, getMemoCache } from '@fluentui-react-native/framework';
+
+const rasterImageStyleCache = getMemoCache<ReactNative.ImageStyle>();
+
+function renderRasterImage(iconProps: IconProps) {
+  const { width, height } = iconProps;
+  const style = rasterImageStyleCache({
+    width: width,
+    height:height
+  }, [width, height])[0];
+
+  return (
+    <ReactNative.Image
+      source = {iconProps.rasterImageSource.src}
+      style = {style}
+    />
+  )
+}
+
+const fontStyleMemoCache = getMemoCache();
+
+function renderFontIcon(iconProps: IconProps) {
+  const fontSource: FontIconProps = iconProps.fontSource;
+
+  const style = fontStyleMemoCache({
+      fontSrcFile: fontSource.fontSrcFile,
+      fontFamily: fontSource.fontFamily,
+      codepoint: fontSource.codepoint,
+      fontSize: fontSource.fontSize,
+      color: iconProps.color
+  }, [iconProps.color, fontSource.fontSrcFile, fontSource.fontFamily, fontSource.codepoint, fontSource.codepoint])[0];
+
+  if (style.fontSrcFile != undefined) {
+    const asset = assetRegistry.getAssetByID(style.fontSrcFile);
+    style.fontFamily = `${style.fontFamily}#${asset.httpServerLocation}/${asset.name}.${asset.type}`;
+  }
+
+  const char = String.fromCharCode(fontSource.codepoint);
+  return(
+    <Text style={style}>{char}</Text>
+  );
+}
+
+function renderSvg(iconProps: IconProps) {
+  const svgIconProps: SvgIconProps = iconProps.svgSource;
+  const { width, height } = iconProps;
+  const viewBox = iconProps.svgSource.viewBox;
+
+  if (svgIconProps.src) {
+    return (
+        <svgIconProps.src
+          viewBox = {viewBox}
+          width = {width}
+          height = {height}
+          color = {iconProps.color}
+        />
+    );
+  }
+  else if (svgIconProps.uri)
+  {
+    return (
+        <SvgUri uri = {svgIconProps.uri}
+          viewBox = {viewBox}
+          width = {width}
+          height = {height}
+          color = {iconProps.color}
+        />
+    )
+  }
+  else {
+    return null;
+  }
+}
+
+
+export const Icon = stagedComponent((props: IconProps) => {
+  const theme = useTheme();
+
+  return (rest: IconProps) => {
+    const color = props.color || theme.colors.buttonText;
+
+    const style = {
+      color: color
```

Comment:
This should be renamed to baseProps or something rather than style since style has a particular meaning.